vfs.go

```diff
@@ -643,3 +644,52 @@ func (v *VFS) List(path string) ([]string, []string, error) {
 
 	return folderData.Folders, folderData.Files, nil
 }
+
+type visitRecursiveCallback = func(path string, size uint64) (doContinue bool, err error)
+
+// visitRecursiveState contains the state of a call to VisitRecursive.
+type visitRecursiveState struct {
+	callback  visitRecursiveCallback
+	lastError error
+}
+
+//export vfsLsRecursive
+func vfsLsRecursive(path *C.cchar_t, path_len C.size_t, size C.uint64_t, data unsafe.Pointer) int32 {
+	state := pointer.Restore(data).(*visitRecursiveState)
+
+	if path_len > math.MaxInt {
+		state.lastError = errors.New("path is too long")
+		return 0
+	}
+
+	doContinue, err := state.callback(C.GoStringN(path, C.int(path_len)), uint64(size))
```

Comment:
Do you mean using `recover()` to catch and rethrow an `error` if the callback panics? IMO if a user writes a callback that panics that's what should happen but @NullHypothesis please correct me if I am wrong.

For example I tried this in `examples/vfs.go` and it panics, I think that's what would be expected.

```golang
err = vfs.VisitRecursive(dirA, func(path string, size uint64) (bool, error) {
    panic("Test")
    return false, nil
})
```

Reply:
Yes it should still panic, but I'm not sure what is the behavior if we panic in a Go function that is called from C code, and whether we should have additional logic to handle it. I know that in C# this is not supported (because there is no standardized ABI for exceptions in non-Windows), but don't know anything about that in Go.